### PR TITLE
Examples and tests for ClusterProcess

### PR DESCRIPTION
**Pull request by ErickChacon**
Tests are currently breaking when working with CartesianGrids and SimpleMeshes due to JuliaGeometry/Meshes.jl#477. Also, let me know if I should remove the MeshViz code.

### CODE REVIEW

File: src/processes/cluster.jl [hunk 1]
```diff
@@ -18,7 +18,58 @@ parent point and returns a geometry or domain for the offspring process.
...
+  parent -> Ball(parent, 0.2)
+)
+ps = rand(p, d)
+
+fig = viz(d, alpha = 0.5; axis = (;title = "Matern cluster process"))
+viz!(ps, color = :black, pointsize = 5)
+
+# Inhomogenegeous parent process with fixed number of offsprings
+
+d = Box((-2.5,-2.5), (2.5,2.5))
```
Comment (juliohm) [inline src/processes/cluster.jl:39]:
```suggestion
```

File: src/processes/cluster.jl [hunk 2]
```diff
@@ -18,7 +18,58 @@ parent point and returns a geometry or domain for the offspring process.
...
+  parent -> Ball(parent, 0.2)
+)
+ps = rand(p, d)
+
+fig = viz(d, alpha = 0.5; axis = (;title = "Inhomogeneneous with fixed number of offsprings"))
+viz!(ps, color = :black, pointsize = 5)
+
+# Inhomogenegeous parent process and inhomogeneneous offspring process
+
+d = Box((0,0), (5,5))
```
Comment (juliohm) [inline src/processes/cluster.jl:52]:
```suggestion
```

File: src/processes/cluster.jl [hunk 3]
```diff
@@ -18,7 +18,58 @@ parent point and returns a geometry or domain for the offspring process.
...
+  parent -> rand(PoissonProcess(x -> 5000 * sum((x - parent).^2)), Ball(parent, 0.5))
+)
+ps = rand(p, d)
+
+fig = viz(d, alpha = 0.5; axis = (;title = "Inhomogenegeous parent and inhomogeneneous offspring"))
+viz!(ps, color = :black, pointsize = 5)
+
+# Inhomogenegeous parent process and regularsampling
+
+d = Box((0,0), (5,5))
```
Comment (juliohm) [inline src/processes/cluster.jl:64]:
```suggestion
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Thank you @ErickChacon for this awesome set of examples and tests. I erased the surrounding code in the docstring to focus on the ClusterProcess constructor. The actual code with MeshViz.jl and other lines can be placed directly in the GeoStats.jl documentation after we merge and tag this.

**Comment by juliohm** [comment]
Tests are failing on this one.

**Comment by ErickChacon** [comment]
> [JuliaGeometry/Meshes.jl#477](https://github.com/JuliaGeometry/Meshes.jl/issues/477).

It is because `view(poinset, CartesianGrid)` is not working (https://github.com/JuliaGeometry/Meshes.jl/issues/477).

**Comment by juliohm** [comment]
Are you sure it used to work? We never intended to support view with Domain
originally, just view(obj, geom::Geometry)

Em seg., 26 de jun. de 2023 12:29, Dr. Erick A. Chacón Montalván <
***@***.***> escreveu:

**Comment by ErickChacon** [comment]
Ok. Then, we need to modify the following to make it work with CartesianGrid and SimpleMesh.

https://github.com/JuliaEarth/PointPatterns.jl/blob/fa3a09f010bd52743f18f78891006fba10c089d1/src/processes/cluster.jl#L43-L45

**Comment by juliohm** [comment]
I think what we need in this case is to compute the intersection of a
PointSet with another Domain. The result is another PointSet or nothing and
we could implement this method in Meshes.intersection. What do you think?

Em seg., 26 de jun. de 2023 12:46, Dr. Erick A. Chacón Montalván <
***@***.***> escreveu:

**Comment by ErickChacon** [comment]
> I think what we need in this case is to compute the intersection of a PointSet with another Domain. The result is another PointSet or nothing and we could implement this method in Meshes.intersection. What do you think? 

Sure! I like the idea.

**Comment by ErickChacon** [comment]
> Are you working on it or should I invest some time to add the method over the week? 

It would be great if you can do it, so I can focus on ther implementations required in `PointPatterns.jl`.

**Comment by juliohm** [comment]
Sounds good. Will take care of it.

Em seg., 26 de jun. de 2023 13:55, Dr. Erick A. Chacón Montalván <
***@***.***> escreveu:

**Pull request merged**